src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -420,26 +420,29 @@ public void Done()
 
             public bool SupportedDiagnosticId(ProjectId projectId, string id)
             {
-                if (_diagnosticIdMap.TryGetValue(projectId, out var ids))
+                lock (_diagnosticIdMap)
                 {
-                    return ids.Contains(id);
-                }
+                    if (_diagnosticIdMap.TryGetValue(projectId, out var ids))
```

Comment:
MultiDictionary looks like itself is not thread safe but value is; by using immutable set for the value. but yet, it doesn't support adding set at once but ask to add value one by one by creating new immutable set repeatedly.

what is its main usage scenario?